src/lib.rs

```diff
@@ -292,6 +293,30 @@ impl From<JsonFrontendError> for Error {
 ///
 /// [`BpfProgram`]: type.BpfProgram.html
 pub fn apply_filter(bpf_filter: BpfProgramRef) -> Result<()> {
+    apply_filter_with_flags(bpf_filter, SeccompFlagset::new())
+}
+
+/// Apply a BPF filter to the all threads in the process.
+///
+/// # Arguments
+///
+/// * `bpf_filter` - A reference to the [`BpfProgram`] to be installed.
+///
+/// [`BpfProgram`]: type.BpfProgram.html
+pub fn apply_filter_all_threads(bpf_filter: BpfProgramRef) -> Result<()> {
+    apply_filter_with_flags(bpf_filter, SeccompFlag::ThreadSync.to_flagset())
+}
+
+/// Apply a BPF filter to the calling thread.
+///
+/// # Arguments
+///
+/// * `bpf_filter` - A reference to the [`BpfProgram`] to be installed.
+/// * `flags` - A u64 representing a bitset of flags, typically created by or-ing [`SeccompFlags`]
+///   together
+///
+/// [`BpfProgram`]: type.BpfProgram.html
+pub fn apply_filter_with_flags(bpf_filter: BpfProgramRef, flags: SeccompFlagset) -> Result<()> {
```

Comment:
considering the various implications of adding flags to the `seccomp` syscall, I wouldn't expose it to the public.
The return type of the function can change depending on the flag (for `SECCOMP_FILTER_FLAG_NEW_LISTENER `, it returns a file descriptor).

For more advanced use cases, let's leave it to the user of the library to install the filters and handle all corner-cases themselves.

The `apply_filter` was meant as a helper for installing filters in the default use-case scenario, when no customisation was needed.